File: server/internal-module/src/main/java/com/example/model/TemplePlayerData.java
package com.example.model;

import com.fasterxml.jackson.annotation.*;

import java.util.*;

public class TemplePlayerData implements PlayerData {

    protected int publicID;
    @JsonIgnore
    private UUID privateID;

    protected String name;
    protected PlayerData.State state;

    @JsonIgnore
    protected String gameName;

    private boolean isGameOwner;

    public TemplePlayerData(){}
    public TemplePlayerData(String name, int publicId) {
        this(name, publicId, false);
    }

    public TemplePlayerData(String name, int publicId, boolean isGameOwner) {
        this.name = name;
        this.publicID = publicId;
        this.isGameOwner = isGameOwner;
    }

    @Override
    public boolean isGameOwner() {
        return isGameOwner;
    }
    
    public void setIsGameOwner(boolean isGameOwner) {
        this.isGameOwner = isGameOwner;

    }
    public int getPublicID() {
        return this.publicID;
    }

    public void setPublicID(int publicID) {
        this.publicID = publicID;
    }

    public UUID getPrivateID() {
        return this.privateID;
    }

    public void setPrivateID(UUID privateID) {
        this.privateID = privateID;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getGameName() {
        return this.gameName;
    }

    public void setGameName(String gameName) {
        this.gameName = gameName;
    }

    public PlayerData.State getState() {
        return this.state;
    }

    public void setState(PlayerData.State state) {
        this.state = state;
    }
    public String toString() {

        return "{" +
                " publicID=" + this.publicID +
                ", name=" + this.name +
                ", state=" + this.state +
                "}";
    }

}

File: UnrealClient/Source/UnrealClient/model/TemplePlayerData.cpp
#include "TemplePlayerData.h"

using namespace std;

TemplePlayerData::TemplePlayerData()
{
}

TemplePlayerData::TemplePlayerData(string name, int publicId) 
		: TemplePlayerData(name, publicId, false)
{
}

TemplePlayerData::TemplePlayerData(string name, int publicId, bool isGameOwner)
{
	this->name = name;
	this->publicID = publicId;
	this->ownsGame = isGameOwner;
}

bool TemplePlayerData::isGameOwner()
{
	return ownsGame;
}

void TemplePlayerData::setIsGameOwner(bool isGameOwner)
{
	this->ownsGame = isGameOwner;

}

int TemplePlayerData::getPublicID()
{
	return this->publicID;
}

void TemplePlayerData::setPublicID(int publicID)
{
	this->publicID = publicID;
}

UUID TemplePlayerData::getPrivateID()
{
	return this->privateID;
}

void TemplePlayerData::setPrivateID(UUID privateID)
{
	this->privateID = privateID;
}

string TemplePlayerData::getName()
{
	return this->name;
}

void TemplePlayerData::setName(string name)
{
	this->name = name;
}

string TemplePlayerData::getGameName()
{
	return this->gameName;
}

void TemplePlayerData::setGameName(string gameName)
{
	this->gameName = gameName;
}

PlayerData::State TemplePlayerData::getState()
{
	return this->state;
}

void TemplePlayerData::setState(PlayerData::State state)
{
	this->state = state;
}

string TemplePlayerData::toString()
{

	//return L"{" + L" publicID=" + std::to_wstring(this->publicID) + L", name=" + this->name + L", state=" + this->state + L"}";
	return "{" + " publicID=" + this->publicID + ", name=" + this->name + ", state=" + this->state + "}";
}


